scripts/floods/earthengine_image.py

```diff
@@ -0,0 +1,598 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to extract geoTIFF image from EarthEngine.
+
+Authenticate EarthEngine/GCloud API with the command, open the URL in a browser
+and paste the token when prompted:
+  earthengine authenticate
+
+If running on a remote machine, authenticate with command:
+  earthengine authenticate --quiet
+then run the gcloud auth command as shown and enter back the token.
+
+To extract flooded regions from the dynamic world dataset, run:
+  python3 earthengine_image.py \
+      --ee_dataset='dynamic_world' `# Pick an existing dataset` \
+      --start_date='2022-10-1' `# Filter dataset for the date range` \
+      --end_date='2022-10-31' \
+      --band='water' `# Pick water band from dynamic world collection` \
+      --band_min=0.7 `# Minimum value for a pixel to be considered flooded` \
+      --reducer='max' `# Get the maximal flood extent over the time period` \
+      --ee_mask='land' `# filter by dataset land to remove permanent water `\
+      --scale=1000 `# Reduce image to 1000m pixels` \
+      |& /tmp/ee.log
+
+This creates a task on Earth Engine.
+To view task status, visit https://code.earthengine.google.com/tasks
+    or run:
+  earthengine task list
+"""
+
+import ast
+import datetime
+import ee
+import glob
+import os
+import sys
+import re
+
+from absl import app
+from absl import flags
+from absl import logging
+from datetime import date
+from datetime import datetime
+from datetime import timedelta
+from dateutil.relativedelta import relativedelta
+
+flags.DEFINE_string('config', '',
+                    'File with configuration parameters as a dictionary.')
+flags.DEFINE_string('ee_dataset', '',
+                    'Load earth engine data set define in config datasets.')
+flags.DEFINE_string('gcs_output', '', 'Prefix for output file names on GCS.')
+flags.DEFINE_string('gcs_project', '', 'GCS project for exporting images.')
+flags.DEFINE_string('gcs_bucket', 'earth_engine_exports',
+                    'GCS bucket for exporting images.')
+flags.DEFINE_string('gcs_folder', '', 'GCS folder for exporting images.')
+flags.DEFINE_string(
```

Comment:
I wonder if all these EE params might be better of as configs rather than flags?